src/java.base/share/classes/java/io/FilterOutputStream.java

```diff
@@ -200,4 +200,13 @@ public void close() throws IOException {
             }
         }
     }
+
+    /**
+     * Tells whether or not this stream is open.
+     *
+     * @return {@code true} if, and only if, this stream is open
+     */
+    protected boolean isOpen(){
```

Comment:
This introduces a new API for subclasses so will require discussion.

Reply:
You can convert this method, or the `closed` field, to be package-private (default access) instead.

Reply:
> I thought of this ,but i was not 100% sure. I will create CSR first.

The current patch changes long standing behavior so that will be the focus of the CSR if this change goes away. I think get agreement on that part first.

Reply:
> I thought of this ,but i was not 100% sure. I will create CSR first.

It's probably a bit premature to create the CSR as there is no agreement here on the API change. I think it would be better to drop this part and focus on the behavior change. As I said, this is a change to long standing behavior that will likely be observed so need to proceed with caution.